lib/trie/db/db.go

```diff
@@ -4,12 +4,35 @@ package db
 
 import (
 	"fmt"
+	"sync"
 
 	"github.com/ChainSafe/gossamer/lib/common"
 )
 
+type Database interface {
+	DBGetter
+	DBPutter
+}
+
+// DBGetter gets a value corresponding to the given key.
+type DBGetter interface {
+	Get(key []byte) (value []byte, err error)
+}
+
+// DBPutter puts a value at the given key and returns an error.
+type DBPutter interface {
+	Put(key []byte, value []byte) error
+}
+
 type MemoryDB struct {
 	data map[common.Hash][]byte
+	l    sync.RWMutex
```

Comment:
I suggest renaming this to mutex.
```suggestion
	mutex    sync.RWMutex
```